lib/ast/expression.ts

```diff
@@ -1191,3 +1191,88 @@ export class ChainExpression extends Expression {
         return [null, newScope];
     }
 }
+
+
+export class JoinExpression extends Expression {
+    lhs : Expression;
+    rhs : Expression;
+    condition : BooleanExpression|null;
+
+    constructor(location : SourceRange|null,
+                left : Expression,
+                right : Expression,
+                condition : BooleanExpression|null, 
```

Comment:
Should condition be Ast.BooleanExpression.True instead of null?

Reply:
fixed in 61a6e46.